src/mame/palm/palm.cpp

```diff
@@ -16,195 +16,442 @@
 #include "machine/mc68328.h"
 #include "machine/ram.h"
 #include "sound/dac.h"
+#include "video/sed1375.h"
+
 #include "emupal.h"
 #include "screen.h"
 #include "speaker.h"
 
 #include "pilot1k.lh"
 
+#define VERBOSE (1)
+#include "logmacro.h"
+
 namespace {
 
-class palm_state : public driver_device
+class palm_base_state : public driver_device
 {
 public:
-	palm_state(const machine_config &mconfig, device_type type, const char *tag) :
-		driver_device(mconfig, type, tag),
-		m_maincpu(*this, "maincpu"),
-		m_ram(*this, RAM_TAG),
-		m_screen(*this, "screen"),
-		m_palette(*this, "palette"),
-		m_io_penx(*this, "PENX"),
-		m_io_peny(*this, "PENY"),
-		m_io_penb(*this, "PENB"),
-		m_io_portd(*this, "PORTD")
+	palm_base_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag)
+		, m_maincpu(*this, "maincpu")
+		, m_ram(*this, RAM_TAG)
+		, m_screen(*this, "screen")
+		, m_io_penx(*this, "PENX")
+		, m_io_peny(*this, "PENY")
+		, m_io_penb(*this, "PENB")
 	{ }
```

Comment:
If this is an abstract base for drivers classes, you can make the constructor protected to make it more obvious.